shpc/main/registry/__init__.py

```diff
@@ -171,18 +175,20 @@ def sync_from_remote(
 
         # No local registry provided, use default
         if not local:
-            local = Filesystem(self.settings.filesystem_registry)
+            local = self.filesystem_registry
 
-        tmpdir = remote.source
-        if tmpdir.startswith("http") or not os.path.exists(tmpdir):
-            tmpdir = remote.clone()
+        need_cleanup = False
```

Comment:
My preference would be to remove this cleanup logic here entirely. The developer is in control. For the sync (which calls this) that is where we clean up for them. As it is now (moving the cleanup to this level) we aren't giving them a choice anymore.